localstack-core/localstack/services/apigateway/integration.py

```diff
@@ -188,7 +188,7 @@ def get_internal_mocked_headers(
 
 
 def get_source_arn(invocation_context: ApiInvocationContext):
-    return f"arn:aws:execute-api:{invocation_context.region_name}:{invocation_context.account_id}:{invocation_context.api_id}/{invocation_context.stage}/{invocation_context.method}{invocation_context.path}"
+    return f"arn:{get_partition(invocation_context.region_name)}:execute-api:{invocation_context.region_name}:{invocation_context.account_id}:{invocation_context.api_id}/{invocation_context.stage}/{invocation_context.method}{invocation_context.path}"
```

Comment:
q: we have the same util in `next_gen`, is it alright if we do a follow PR to implement this change as well? in `localstack.services.apigateway.next_gen.execute_api.helpers.get_source_arn`. If not important we can change it in a follow PR for us too. 😄

Reply:
Very good catch! I would do it in an immediate follow up, I would like to get this one in as soon as possible to avoid merge conflicts (rebasing this one is a real pain), and fix not-critical issues in follow ups. If there is any other change necessary anyway, I will make it here!